lib/views/resetPassword.js

```diff
@@ -56,17 +56,47 @@ function resetPasswordPost (req, res) {
 
     return user.save()
   }).then((user) => {
-    sendResetPasswordMail(user)
+    return sendResetPasswordMail(user)
+  }).then(() => {
     if (!req.accepts('text/html')) {
-      return res.status(200).header('content-type', 'text/plain').send('Password reset successfully')
+      return res.status(200).header('content-type', 'text/plain').send('If a user was found associated with the specified e-mail address, a link to reset your password has been sent.')
     } else {
       res.send(pug.renderFile('templates/views/resetPasswordDone.jade', locals))
     }
   }).catch((err) => {
-    if (!req.accepts('text/html')) {
-      return res.status(500).header('content-type', 'text/plain').send(err)
+    if (err === {}) {
+      if (!req.accepts('text/html')) {
+        return res.status(200).header('content-type', 'text/plain').send('If a user was found associated with the specified e-mail address, a link to reset your password has been sent.')
+      } else {
+        res.send(pug.renderFile('templates/views/resetPasswordDone.jade', locals))
+      }
     } else {
-      res.send(pug.renderFile('templates/views/resetPasswordDone.jade', locals))
+      var error = err
+      var statusCode = 500
```

Comment:
I think you can just do `statusCode = err.response.statusCode || 500`

Reply:
What if err is undefined?  Would this not create an exception?  The logic is complex because I was not entirely sure what I would be getting back.  This is a error returned by Sendgrid.  I was being very particular, since I was parsing the JSON coming back, and I think assuming and dereferencing can lead to exceptions.  I'm not sure though about the syntax you give though if that is more robust to dereference problems.

Reply:
I just tested, and it will create an exception. Can be left as-is.